src/battle_script_commands.c

```diff
@@ -8324,7 +8289,7 @@ static bool32 IsAbilityAbsorbAffected(void)
     else
         moveType = gBattleMoves[gCurrentMove].type;
 
-    if (moveType == TYPE_ELECTRIC && GetBattlerAbility(gBattlerTarget) == ABILITY_VOLT_ABSORB)
+    if (moveType == TYPE_ELECTRIC && GetBattlerAbility(gBattlerTarget) == ability && gBattleMoves[gCurrentMove].effect != EFFECT_ION_DELUGE)
```

Comment:
Maybe target `MOVE_TARGET_ALL_BATTLERS` would be better instead of the effect so that it applies to all future moves with a similar effect?

Reply:
You're absolutely right, this also works for Electric Terrain which my fix didn't include.

Reply:
Tests fail. I guess it's because it also needs to check for type?

Reply:
Alright, should be good now